File: backend/src/CreditFinancialOrganization.Domain/Entities/Loans/Loan.cs
using CreditFinancialOrganization.Domain.Entities.Users;
using CreditFinancialOrganization.Domain.Enums;

namespace CreditFinancialOrganization.Domain.Entities.Loans;

public class Loan : BaseEntity
{
    public Guid CustomerId { get; set; }
    public Guid LoanTypeId { get; set; }
    public LoanStatus Status { get; set; }
    public decimal Amount { get; set; }
    public decimal InterestRate { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }

    public User Customer { get; set; } = null!;
    public LoanType LoanType { get; set; } = null!;
    public LoanApplication Application { get; set; } = null!;
}

File: frontend/src/app/shared/models/loans/loan.ts
import { LoanStatus } from "../../enums/loan-status";
import { User } from "../users/user";
import { LoanType } from "./loan-type";

export interface Loan {
    id: string;
    customer: User;
    loanType: LoanType;
    status: LoanStatus;
    amount: number;
    interestRate: number;
    startDate: Date;
    endDate: Date;
}